### PR TITLE
[16.0][MIG] product_internal_reference_generator

### PR DESCRIPTION
**Pull request by sbiosca-s73**
Migration module product_internal_reference_generator to v16.0

### CODE REVIEW

**Comment by sbiosca-s73** [comment]
@pedrobaeza please apply the "no stale" label in this PR.

**Comment by pedrobaeza** [comment]
Well, instead, enroll reviewers for it.

/ocabot migration product_internal_reference_generator

**Comment by sbiosca-s73** [comment]
@pedrobaeza please put "no stale" label. @ilyasProgrammer can you review the code?

**Review by ilyasProgrammer — APPROVED** [review]
LGTM

File: product_internal_reference_generator/tests/test_product_internal_reference_generator.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 Ooops - Ilyas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.tests import Form, TransactionCase
+
+
+class TestProductInternalReferenceGenerator(TransactionCase):
```
Comment (rousseldenis) [inline product_internal_reference_generator/tests/test_product_internal_reference_generator.py:6]: You could maybe use `BaseCommon`class from https://github.com/odoo/odoo/blob/a877336b4d007c06c9759e4c8a302038bae20a26/odoo/addons/base/tests/common.py#L19

As it enhances tests performances.

File: product_internal_reference_generator/tests/test_product_internal_reference_generator.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    @classmethod
+    def setUpClass(cls):
+        super(TestProductInternalReferenceGenerator, cls).setUpClass()
```
Comment (rousseldenis) [inline product_internal_reference_generator/tests/test_product_internal_reference_generator.py:9]:
```suggestion
        super().setUpClass()
```

**Review by rousseldenis — APPROVED** [review]
Minor remark

**Review by dreispt — APPROVED** [review]

**Pull request merged**